main/pipeline/pipeline.cc

```diff
@@ -173,8 +173,28 @@ const unique_ptr<parser::Node> convertPrismToSorbet(pm_node_t *node, pm_parser_t
             return make_unique<parser::Integer>(locOffset(loc, parser), std::to_string(intNode->value.value));
         }
         case PM_PROGRAM_NODE: {
-            pm_statements_node *stmts = (reinterpret_cast<pm_program_node *>(node))->statements;
-            return convertPrismToSorbet((pm_node *)stmts, parser, gs);
+            pm_program_node *programNode = reinterpret_cast<pm_program_node *>(node);
+            pm_statements_node *stmts = programNode->statements;
+
+            auto size = stmts->body.size;
+
+            // For a single statement, do not create a Begin node and just return the statement
+            if (size == 1) {
+                return convertPrismToSorbet((pm_node *)stmts->body.nodes[0], parser, gs);
+            }
+
+            // For multiple statements, convert each statement and add them to the body of a Begin node
+            parser::NodeVec sorbetStmts;
+
+            for (int i = 0; i < stmts->body.size; i++) {
+                pm_node_t *node = stmts->body.nodes[i];
+                unique_ptr<parser::Node> convertedStmt = convertPrismToSorbet(node, parser, gs);
+                sorbetStmts.emplace_back(std::move(convertedStmt));
+            }
+
+            auto *loc = &programNode->base.location;
+
+            return make_unique<parser::Begin>(locOffset(loc, parser), std::move(sorbetStmts));
```

Comment:
I think this is fine if you think adding a node will require too many changes.

My C++ comment would be to iterate using a range instead:
`for (auto node : stmts->body.nodes) {}`.  It's cleaner and prevents range bugs. This version calls the copy constructor for `node` creation which may be inefficient and [typed differently](https://stackoverflow.com/a/15927037) depending on what you need it for.

Reply:
It's really good to know you can do that in C++! I actually can't iterate over a `pm_node_list` this way because it doesn't implement the `begin` function (I get the error `Invalid range expression of type 'struct pm_node **'; no viable 'begin' function available`). I can look into adding that to the Prism API, but for now I think this is the only way to iterate.